include/tinyalsa/mixer.h

```diff
@@ -127,10 +127,14 @@ int mixer_ctl_get_value(const struct mixer_ctl *ctl, unsigned int id);
 
 int mixer_ctl_get_array(const struct mixer_ctl *ctl, void *array, size_t count);
 
+int mixer_ctl_get_byte_array(const struct mixer_ctl *ctl, void *array, size_t count);
```

Comment:
I see that you're adding a new API, but even though it's called mixer_ctl_get_byte_array, the new API is a copy of the existing mixer_ctl_get_array, with the tlv bug fix. It can still handle all types of array (integer, boolean, etc). With the changes to tinymix, now the existing APIs are never used.

Instead, can you just fix the existing APIs?

Reply:
Thanks Simon for reviewing.
I did not updated exisiting API as I thought there might be tinyalsa clients (other than tinymix) which would still be using the API. Currently mixer_ctl_get./set_array() adds TLV header in the API itself. Considering that, tinymix and other clients should not add any TLV header when calling this API.

What fix do you suggest in the existing API? We can either handle length check in these API and update tinymix to not add header or vice-versa.

Reply:
IMO it's a bad idea to rely on the calling application (like tinymix) to know a detail like the tlv header, when the API itself makes no mention of it. The APIs for mixer_ctl_get/set_array just require an array and a count. So my suggestion would be to keep only these existing APIs and fix them so that they work transparently for byte controls, and make the corresponding changes to tinymix too.

BTW, I noticed that TLV_HEADER_SIZE is problematic. First, it's defined as 2 * sizeof(unsigned int) instead of the struct size itself. Second, it's added to the num_values variable in tinymix_detail_control, which is wrong (it's a size not a count). I can file a new issue on this.

Reply:
Right . I have updated mixer.c and tinymix.c to handle this. Please review.